src/main/java/net/snowflake/client/jdbc/telemetry/TelemetryClient.java

```diff
@@ -287,9 +287,12 @@ private boolean sendBatch() throws IOException
     {
       //session shared with JDBC
       String sessionToken = this.session.getSessionToken();
+      String payload = logsToString(tmpList);
+
+      logger.debugNoMask("Payload of telemetry is : " + payload);
```

Comment:
Question - We are doing this in order for the telemetry to get dumped to the logs and our regression tests can scan for credentials. But in the process we would have dumped actual credentials (if there were any) to user's logs as well right? Can we guard this behavior in some way that ONLY when regression tests are running we call debugNoMask else we call the regular debug logging API?

Reply:
My concern here is if we use secret detector for dumping payload to customer's log, the payload would be over masking because of the lack of delimiter (e.g. space). What if we mask telemetry payload when it's built and keep using debugNoMask here? I think this can reach both of our goals. What do you think?

Reply:
But yeah, we can use ways like environment variable to judge whether we are on jenkins and to choose `debug` or `debugNoMask`. And after I did little investigation on my previous idea, this solution is way more easier than my thought above.

Reply:
> My concern here is if we use secret detector for dumping payload to customer's log, the payload would be over masking because of the lack of delimiter (e.g. space). What if we mask telemetry payload when it's built and keep using debugNoMask here? I think this can reach both of our goals. What do you think?

I did not completely understand this.